noir/noir-repo/acvm-repo/acir/src/circuit/opcodes/black_box_function_call.rs

```diff
@@ -93,11 +105,11 @@ pub enum BlackBoxFuncCall {
         /// is more than the number of bytes in the input,
         /// then an error is returned.
         var_message_size: FunctionInput,
-        outputs: Vec<Witness>,
+        outputs: Box<[Witness; 32]>,
     },
     Keccakf1600 {
-        inputs: Vec<FunctionInput>,
-        outputs: Vec<Witness>,
+        inputs: Box<[FunctionInput; 25]>,
+        outputs: Box<[Witness; 25]>,
```

Comment:
This does add a hard requirement on fields being able to store a u64 in a single witness - keeping these as vectors would allow us to split the lanes over multiple witnesses if necessary.

If we got down to field sizes that small then we're going to run into a lot of problems however.